query_execution/Foreman.hpp

```diff
@@ -522,6 +525,7 @@ class Foreman : public Thread {
   WorkerDirectory *workers_;
 
   friend class ForemanTest;
+  FRIEND_TEST(ForemanTest, TwoNodesDAGPartiallyFilledBlocksTest);
```

Comment:
`TwoNodesDAGPartiallyFilledBlocksTest` accesses `Foreman`'s private member `query_context_`  to mock to add partially filled blocks in `InsertDestination`.